tests/components/subaru/conftest.py

```diff
@@ -83,23 +93,21 @@ def advance_time_to_next_fetch(hass):
 
 async def setup_subaru_integration(
     hass,
+    mock_config_entry=None,
     vehicle_list=None,
     vehicle_data=None,
     vehicle_status=None,
     connect_effect=None,
     fetch_effect=None,
 ):
     """Create Subaru entry."""
-    assert await async_setup_component(hass, HA_DOMAIN, {})
-    assert await async_setup_component(hass, DOMAIN, {})
-
-    config_entry = MockConfigEntry(
-        domain=DOMAIN,
-        data=TEST_CONFIG,
-        options=TEST_OPTIONS,
-        entry_id=1,
-    )
-    config_entry.add_to_hass(hass)
+    if mock_config_entry:
```

Comment:
Please don't add any conditional logic in tests. Just inline the calls in the tests or use pytest fixtures to compose the test setup code.